cfme/markers/env_markers/appliance.py

```diff
@@ -0,0 +1,112 @@
+import random
+
+import pytest
+
+from cfme.markers.env import EnvironmentMarker
+from cfme.markers.env_markers import _param_check
+from cfme.utils import testgen
+from cfme.utils.log import logger
+from cfme.utils.pytest_shortcuts import fixture_filter
+
+
+def parametrize(metafunc, argnames, argvalues, *args, **kwargs):
+    """parametrize wrapper that calls :py:func:`_param_check`, and only parametrizes when needed
+
+    This can be used in any place where conditional parametrization is used.
+
+    """
+    if _param_check(metafunc, argnames, argvalues):
+        metafunc.parametrize(argnames, argvalues, *args, **kwargs)
+    # if param check failed and the test was supposed to be parametrized around a provider
+    elif 'appliance' in metafunc.fixturenames:
+        try:
+            # hack to pass trough in case of a failed param_check
+            # where it sets a custom message
+            metafunc.function.uncollect
+        except AttributeError:
+            pytest.mark.uncollect(
+                reason="provider was not parametrized did you forget --use-provider?"
+            )(metafunc.function)
+
+
+def appliances(metafunc, app_types, fixture_name='appliance'):
+    """ Gets appliances based on given (+ global) filters
+
+    Note:
+        Using the default 'function' scope, each test will be run individually for each appliance
+        before moving on to the next test. To group all tests related to single appliance together,
+        parametrize tests in the 'module' scope.
+
+    Note:
+        testgen for providers now requires the usage of test_flags for collection to work.
+        Please visit http://cfme-tests.readthedocs.org/guides/documenting.html#documenting-tests
+        for more details.
+    """
+    argnames = []
+    argvalues = []
+    idlist = []
+
+    # supported_providers are the ones "supported" in the supportability.yaml file. It will
+    # be a list of DataProvider objects and will be filtered based upon what the test has asked for
+    holder = metafunc.config.pluginmanager.get_plugin('appliance-holder')
+    current_appliance = holder.held_appliance
+    argnames.append(fixture_name)
+    for app_type in app_types:
+        idlist.append(app_type)
+        if app_type == current_appliance.type:
+            argvalues.append([current_appliance])
+        else:
+            msg = (f'No appliance of such type available. needed {app_type}, '
+                   f'exists {current_appliance.type}')
+            argvalues.append(pytest.param(None, marks=pytest.mark.skip(reason=msg)))
+    return argnames, argvalues, idlist
+
+
+class ApplianceEnvironmentMarker(EnvironmentMarker):
+    NAME = 'appliance'
+    DEFAULT = 'regular'
```

Comment:
Not sure I like using 'regular' here, maybe `'default'` instead?  Maybe `'virt'` for virtualized?

Reply:
Maybe just `'single'`? Obviously there isn't an obvious answer.

Reply:
@ManageIQ/qe-triage  and @ManageIQ/qe-committers  Any thoughts here, for what we call our 'normal', 'regular', 'default' appliance type?

Reply:
`default` has my vote

Reply:
my vote for `default` as other options are `multi-region`, `dummy`, `dev`, `pod`, `upgraded`.